woocommerce/google-listings-and-ads
# Add API endpoints to update and fetch contact information

### Changes proposed in this Pull Request:



Part of https://github.com/woocommerce/google-listings-and-ads/issues/863. 

This is pointed to the `feature/contact-info` branch (so that we don't merge unfinished features to trunk) but that branch doesn't seem to contain the latest `trunk` changes so most of the changes here seem new but they're mainly based on Justin's PR https://github.com/woocommerce/google-listings-and-ads/pull/896.

The previously added API endpoints for updating and fetching the phone number are now modified to also handle updating the business address.

The endpoint name is changed from `phone-number` to `contact-information` but it still only accepts one `phone_number` input. However, now it also updates the address in Merchant Center to the store's address set in WooCommerce settings.

The `GET` endpoint now also returns three extra parameters in its response:
- `mc_address`: The address stored in Merchant Center
- `wc_address`: The address stored in WooCommerce Settings
- `is_mc_address_different`: A boolean indicating whether the two addresses above are different

This also sets the country for the Merchant Center account but given that the MC accounts created by GLA are all sub-accounts and [Google does not allow changing a sub-account's country](https://developers.google.com/shopping-content/reference/rest/v2.1/accounts#AccountAddress) we might need to modify this once the Google API behavior is changed. Google API currently doesn't enforce this and allows modifying the country (even though it's not possible to do so via MC interface).

Merchant Center API only accepts one address line and to adapt the WooCommerce address to it, the two addresses entered in WooCommerce settings are merged into one string separated with a new line character (`\n`). MC interface detects the addresses correctly and displays each in their own fields.

Note that address components are returned in the API response as they appear on Google API. This means that their naming might be different from WooCommerce. For example, city is called `locality`, state is called `region`, etc. This is just for the sake of simplicity and can be modified if we want to keep things similar to WC core.


### Detailed test instructions:

1. Connect a Merchant Center account and set a physical address for your store in WooCommerce settings.
2. Make a request to the `GET /wc/gla/mc/contact-information` endpoint
    - Confirm that no phone number or address is returned if you don't have anything in MC
    - if you already have a phone number and address set in MC make sure that it is returned correctly
    - Confirm that the `wc_address` contains the same address as the one stored in WooCommerce settings
3. Update the phone number using the `POST /wc/gla/mc/contact-information` endpoint
   - Confirm that the phone number is updated using the above `GET` endpoint
   - Confirm that the phone number is updated in Merchant Center : 
     <https://merchants.google.com/mc/merchantprofile/businessinfo>
   - Confirm that the address is updated in Merchant Center and it's the same as your WooCommerce address


### Changelog entry
